### PR TITLE
Add ebs__enum_snapshots_unauth Module for Unauthenticated EBS Snapshot Reconnaissance

### PR DESCRIPTION
**Pull request by Y4nush**
This pull request introduces a new module, ebs__enum_snapshots_unauth, designed for unauthenticated reconnaissance of public EBS snapshots. The module offers users the capability to search for snapshots based on account-id, keyword, list of account-ids, or a list of keywords. The purpose of this feature is to uncover snapshots related to a specific organization, thereby aiding in further reconnaissance.

Two main approaches are supported by this module:

1. Account ID Enumeration: This approach is useful when an organization's account ID is disclosed. It allows for the enumeration of snapshots related to the organization.
2. Keyword Enumeration: This method is employed to find snapshots related to an organization through the search of related keywords, aiming to disclose the organization's account ID.

This module was inspired by the article [Assume the Worst: AWS Assume Role Enumeration](https://rhinosecuritylabs.com/aws/assume-worst-aws-assume-role-enumeration/), and it aims to complement the existing ebs__enum_volumes_snapshots module, which appears to be more intended for authenticated enumeration.

Thank you for considering this addition. Looking forward to any feedback or suggestions for improvement.

### CODE REVIEW

**Comment by DaveYesland** [comment]
Hi, thanks for the pull request. 
This module looks like a great idea. Sorry for the delay but we will review this as soon as possible! appreciate the contribution!

File: pacu/modules/ebs__enum_snapshots_unauth/main.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        print("No regions found. Exiting...")
+        return
+
+    session_name = session.name
+    directory = str(downloads_dir() / session_name / 'ebs')
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    for region in regions:
+        print('[ebs__enum_snapshots_unauth] Starting region {}...'.format(region))
```
Comment (DaveYesland) [inline pacu/modules/ebs__enum_snapshots_unauth/main.py:98]: remove the preceding "[ebs__enum_snapshots_unauth] " this is already handled by Pacu print()

File: pacu/modules/ebs__enum_snapshots_unauth/main.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        for keyword in keyword_list:
+            filters = [{'Name': 'status', 'Values': ['completed']}, {'Name': 'description', 'Values': [f'*{keyword}*']}]
+            snapshots = search_snapshots(ec2Client, filters)
+            for snapshot in snapshots:
+                snapshot['Region'] = region
+                snapshot['Keyword'] = keyword
+                snapshot.setdefault('AccountId', '')
+                snapshot.setdefault('Description', '')
+                print('[+] Snapshot found: {}'.format(snapshot['SnapshotId']))
```
Comment (DaveYesland) [inline pacu/modules/ebs__enum_snapshots_unauth/main.py:109]: remove the preceding "[+] " Pacu's print already has a formatted print()

**Review by DaveYesland — CHANGES_REQUESTED** [review]
Just a couple formatting changes and I can get this merged. Let me know if you have an questions.

**Comment by Y4nush** [comment]
Hi Dave, thanks for your review. I have fixed the issues.

**Review by DaveYesland — APPROVED** [review]

**Pull request merged**